### ISSUE TITLE
Android Navigation Bar covers bottom of quest dialogs

### ISSUE THREAD

**Issue by peternewman** [westnordost/StreetComplete#1977]
Android Navigation Bar covers bottom of quest answers
![Screenshot_20200721-015220](https://user-images.githubusercontent.com/6216436/88000157-e588cc00-caf4-11ea-8f9a-31e0d742c1a5.png)

**How to Reproduce**
Answer a quest

**Versions affected**
v21.2 (although it's been happening on earlier versions, I just failed to report it) Android 9

**Comment by matkoniecz** [comment]
Vespucci has option to not use this part of the screen - see https://github.com/MarcusWolschon/osmeditor4android/issues/955

**Comment by westnordost** [comment]
@peternewman Well, what do you propose? you can pull the form up, you know?

**Comment by peternewman** [comment]
> Vespucci has option to not use this part of the screen - see [MarcusWolschon/osmeditor4android#955](https://github.com/MarcusWolschon/osmeditor4android/issues/955)

Thanks @matkoniecz that had been annoying me on Vespucci too, making fullscreen never seems to work.

> @peternewman Well, what do you propose?

I'd like the interface to be fully visible by default, so if I'm answering this (or particularly tactile paving), I can just tap to select the quest and tap again to answer, I don't have to get lucky with my taps or drag things up and down.

SC seems to be one of the few apps I have which shows anything below the navigation bar, I can sort of see the benefit of seeing a bit more map, but everything else just crops it's UI to fit the available screen space (excluding the navigation bar).

Interestingly Vespucci only seems to trip up for me when I try and move between menus, so preferences works for example, so perhaps it's fullscreen mode would work for SC as I rarely select one bottom option and then change my mind and select another.

> you can pull the form up, you know?

I'm not sure I particularly knew, I think I normally just jab at the bottom and I get it after one or two goes.

If you wanted to make it smarter/better, I could see the benefit of a compromise so if the bottom row just contains Can't Say/Other Answers, it could be tucked below the Navigation Bar (similarly although unnecessary in this case given the other options with the surface quests I wouldn't object if show more was hidden down), but it would be good if all frequent options were accessible without scrolling (the half of the surface images is sufficient for that), the 1/3 of the yes/no for tactile paving is too intermittent in my experience.

**Comment by westnordost** [comment]
It currently works like that the bottom sheet (that's how these kind of pull-up-forms are named) expands to a certain max height automatically. If the total height of the form is higher than that certain max height, the user will have to pull up the rest himself.

Most of the "select one of these icons"-quests (sport, surface, ...) are too high to fit, so you need to pull it up for more choices.

So, a few quests (in certain languages) are just above the max height, so you always need to pull them up by just a little bit.

**Comment by peternewman** [comment]
> It currently works like that the bottom sheet (that's how these kind of pull-up-forms are named) expands to a certain max height automatically. If the total height of the form is higher than that certain max height, the user will have to pull up the rest himself.

Could we make that configurable (by the user)? Equally in landscape mode, the max height is presumably greater than the height of my screen, so it obscures most of the map anyway. This seems to be the only disadvantage of having a greater max height.

Therefore could we slightly increase that max height on smaller DPI screens.

> Most of the "select one of these icons"-quests (sport, surface, ...) are too high to fit, so you need to pull it up for more choices.
> 
> So, a few quests (in certain languages) are just above the max height, so you always need to pull them up by just a little bit.

This is only the tactile, so it's just illustrative images, interestingly the bus stop fits, it's actually only the wording that runs to three lines causing it to not fit.

**Comment by peternewman** [comment]
![Screenshot_20200722-133134](https://user-images.githubusercontent.com/6216436/88178134-cfc4f500-cc21-11ea-9663-8631f1bb648d.png)

**Comment by westnordost** [comment]
> Therefore could we slightly increase that max height on smaller DPI screens.

The max height is relative to the DPI.

> There seems to be a lot of whitespace to the right hand side of the quest bubble here. If that could be removed/reduced this particular issue would probably go away.

There is an invisible X-button on the upper right corner that would become visible if the form covers the whole screen vertically

**Comment by peternewman** [comment]
> > Therefore could we slightly increase that max height on smaller DPI screens.
> 
> The max height is relative to the DPI.

Do you mean across the whole range, or stepped? The Sony Xperia XZ1 is 720x1280 and 319 DPI apparently.

> > There seems to be a lot of whitespace to the right hand side of the quest bubble here. If that could be removed/reduced this particular issue would probably go away.
> 
> There is an invisible X-button on the upper right corner that would become visible if the form covers the whole screen vertically

Is it not possible to also reduce the width of that extra padding when the button is invisible?

I assume you mean this X? This is the only road name I could find that fits. Silver is too long.

**Comment by westnordost** [comment]
The "peek height" (max initial height) is 352dip. So for devices that have about 3x the dpi as a normal computer screen, this will be 1056px and so on.

>  Silver is too long.

What silver?

**Comment by peternewman** [comment]
> The "peek height" (max initial height) is 352dip. So for devices that have about 3x the dpi as a normal computer screen, this will be 1056px and so on.

It's about half the screen for me.

I've done 261 of these quests according to the stats, so it is a genuine irritation if it doesn't work first time.

> > Silver is too long.
> 
> What silver?

Sorry, that was me testing street names, I tried a Silver Street, and the whole name wrapped, which seems to me like a lot of padding around the X. But as I say, removing the width of it's spacing when it's not visible, if possible, seems the perfect solution to me.

**Comment by westnordost** [comment]
Ok, I did the following now:

- the close button is initially GONE so the question text has more space
- for bigger displays (720dp in height and up), higher peek height. Devices that have a display ratio of slimmer than 16 : 9, for example 16 : 10 or 18.5 : 9 should fall into that category. Pretty much all modern devices migrated from a display ratio of 16 : 9 to something slimmer/longer. The new "compact" series of Sony (Xperia 5) even went towards longer than 21 : 9.

**Comment by peternewman** [comment]
Thanks @westnordost !

I'm certainly aware my screen is relatively short and fat as a lot of media I get sent gives me pillarbox black edges because it's designed for a taller screen.

**Comment by westnordost** [comment]
What model do you use?

**Comment by peternewman** [comment]
> What model do you use?

> The Sony Xperia XZ1 is 720x1280 and 319 DPI apparently.

**Comment by westnordost** [comment]
I see, so 360x640dp

**Comment by peternewman** [comment]
If you say so, I struggled to find an easy way to find out that value. Sounds like I won't benefit from half your fix then, but hopefully GONE is enough. :crossed_fingers:

**Comment by westnordost** [comment]
Yeah, you know, I still got a Sony Xperia Z1 Compact.
The waterproofing is broken, it's the third battery, the back cover is completely gone, the NFC chip has peeled off and the camera became blurry/unfocussed so I am looking into it replacing it in a few months.

**Comment by peternewman** [comment]
Glad to hear they are that well built! For me having a headphone jack was quite a strong driver too, as well as being compact.

**Comment by westnordost** [comment]
For me too, at least relatively important. All of the Samsung A41 (220€), Samsung S10e (520€) and Pixel 3a (270€) have a headphone jack. Pixel 4a hasn't been announced yet.

What's also important especially if you plan to use the phone for a long time is that the company has a proper update policy (Sony didn't, stuck on Android 5.1) and that it has a good repairability, in particular that the battery can be changed more or less easily. Of the ones I looked at, Pixel 3a has a good repairability, Pixel 4 and Samsung S10e for example not at all.

Anyway, if you are interested, I made a chart of compact phones I'd maybe buy. Xiaomi, Redmi, OnePlus and maybe Huawai etc. are unfortunately out of the race because of privacy concerns. (Apparently they record usage data and whatnot without consent)


[smartphones.xlsx](https://github.com/westnordost/StreetComplete/files/4972416/smartphones.xlsx)
